zerver/models/messages.py

```diff
@@ -232,6 +233,11 @@ class Meta:
                 F("id").desc(nulls_last=True),
                 name="zerver_message_realm_id",
             ),
+            models.Index(
+                "realm_id",
+                "type",
+                name="zerver_message_realm_type",
+            ),
```

Comment:
Hmm, I wonder if a partial index just on cases where it isn't `NORMAL` would be most correct?

Reply:
We now use a new table as per our discussion in https://chat.zulip.org/#narrow/stream/378-api-design/topic/message.20type.20parameter.2C.20.2324960/near/1839346